index.js

```diff
@@ -114,17 +116,27 @@ function recursivelyIterateProperties(jsonObject) {
     jsonObject.repeat = undefined;
     jsonObject.structuredExample = undefined;
 
+
     if (jsonObject.type === 'object' && _.isObject(jsonObject.properties)) {
         // Find all required properties
         jsonObject.required = _.map(_.filter(jsonObject.properties, 'required'), 'name');
 
         // Parse children properties
         _.forEach(jsonObject.properties, function (propObject, propKey) {
-            recursivelyIterateProperties(propObject)
+            recursivelyIterateProperties(types, propObject)
         })
     } else {
         jsonObject.required = undefined;
     }
+
+    var typeName = jsonObject.type
+    if (_.indexOf(types, typeName) >= 0) {
+        _.forEach(_.keys(jsonObject), function(key) {
+            delete jsonObject[key]
+        })
+
+        jsonObject["$ref"] = "file://build/schema/" + typeName + "#"
```

Comment:
We're actually taking the `"file://build/schema/"` from the params (as `program.prefix`). It would require passing one more thing through, but using that means that this would still work if we ever changed where API_Spec dumps the generated schemas.